### PR TITLE
Add reqs that needs to be installed before test_requirements.txt

### PR DESCRIPTION
**Pull request by Xarthisius**
## PR Summary

This PR introduces additional file (`tests/test_prerequirements.txt`) that can be used to install basic packages required by anything in `test_requirements.txt` (I'm looking at you Cartopy...). It allows me to remove hardcoded deps that are currently defined in the Jenkins' job. I also dropped pinning mpl to 3.1.3, so there's a high chance this will fail and will require some answers to be regenerated.

### CODE REVIEW

**Comment by Xarthisius** [comment]
I bumped the answers. Apart from one case when we were plotting an empty image (fixed in 037a92b) all the failures are due to slight changes in ticks labels position, e.g.:

tmpiin7wj9a-failed-diff

More failures can be found here: https://tests.yt-project.org/job/yt_py38_git/1363/#showFailuresLink

File: tests/test_prerequirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# We need this file mostly because of Cartopy..
```
Comment (munkm) [inline tests/test_prerequirements.txt:1]: yeah the build was super messy before we did this

**Review by munkm — APPROVED** [review]

**Comment by munkm** [comment]
Maybe it is worth updating the CI scripts to use the test_prerequirements file too?

**Review by neutrinoceros — APPROVED** [review]
LGTM. Is this ready to go ?

**Pull request merged**